### DISCUSSION TITLE
Actioncontroller::InvalidAuthenticityToken with omniauth_openid_connect and omniauth-rails_csrf_protection

### DISCUSSION THREAD

**Discussion by mchavarriagam** [omniauth/omniauth#1130] [Q&A]
Hello,

I'm not sure if this is the right place to ask about this, so apologies in advance.

Running Rails v5.2.6 with Ruby v2.5.8, omniauth v2.1.2, omniauth-rails_csrf_protection v1.0.1, and omniauth_openid_connect v0.6.1

When a third party POST request comes into our OIDC auth endpoint (/auth/openid_connect), omniauth starts the request validation phase (https://github.com/omniauth/omniauth/blob/master/lib/omniauth/strategy.rb#L240) which ultimately fails in omniauth-rails_csrf_protection with verified_request? being false (https://github.com/cookpad/omniauth-rails_csrf_protection/blob/main/lib/omniauth/rails_csrf_protection/token_verifier.rb#L33).

I assume(?) that this being a third party POST request into our OIDC auth would mean no valid request authenticity tokens would be sent (if this is the case then no request validation phase should take place for this specific request?), but I'm not sure how to prevent the request validation phase from being triggered. My assumption may also be incorrect, in which case I'm not sure how to proceed.

Any ideas? Thanks in advance.

**Comment by BobbyMcWho** [comment]
POST requests typically come from your own application's login button, and the callback from the provider comes in via GET

**Reply by mchavarriagam** [reply]
Not in this case @BobbyMcWho. This is part of an LTI 1.3 request, so the flow begins at the "Platform" (the third party), which is pointed to our OIDC auth endpoint as a POST request. This in turn initiates the OIDC auth flow. This is a good walk through of the process -> https://andyfmiller.com/2018/12/28/launching-an-lti-1-3-resource-link-using-openid-connect-third-party-login/

Now, I understand what you're saying regarding "(...)typically come from your own application(...)" which is why I'm wondering if in this use case (POST request originated from external platform) the validation phase should not be triggered. I'm not sure how to go about it though.

**Reply by BobbyMcWho** [reply]
Are you using the OIDC connect strategy? https://github.com/omniauth/omniauth_openid_connect?tab=readme-ov-file

Edit: sorry the title indicates you are
